File: backend/src/main/java/fr/formation/backend/model/Transmission.java
package fr.formation.backend.model;

public enum Transmission {
    orale("Orale"), sexuelle("Sexuelle"), sang("Contact sanguin"), direct("Contact direct"), materno("Materno-fœtale");

    private String typeTransmission;

    private Transmission(String typeTransmission) {
        this.typeTransmission = typeTransmission;
    }

    public String getTypeTransmission() {
        return typeTransmission;
    }

    public void setTypeTransmission(String typeTransmission) {
        this.typeTransmission = typeTransmission;
    }

}

File: frontend/src/app/enum/transmission.ts
export type Transmission = 'orale' | 'sexuelle' | 'sang' | 'direct' | 'materno';

export const TransmissionEmoji: Record<Transmission, { emoji: string; label: string }> = {
  orale: { emoji: '👄', label: 'Orale' },
  sexuelle: { emoji: '🍆', label: 'Sexuelle' },
  sang: { emoji: '💉', label: 'Contact sanguin' },
  direct: { emoji: '🤝', label: 'Contact direct' },
  materno: { emoji: '👶', label: 'Materno-fœtale' },
};
